### PR TITLE
Simpler null parsing

### PR DESCRIPTION
**Pull request by arj03**
I noticed that null was generating way too much code. This cleans things up a bit. Not 100% about the fullMatch thing.

### CODE REVIEW

**Comment by mafintosh** [comment]
@arj03 full match means it'll match the null string fully. you need to check that the next 4 chars are n,u,l,l in that one.

You can use the `cmp` helper for that, https://github.com/mafintosh/turbo-json-parse/blob/master/lib/ops.js#L47

**Comment by mafintosh** [comment]
Sorry disregard that. This is fine.

File: lib/null.js [hunk 1]
```diff
@@ -12,38 +12,26 @@ function defaults (opts) {
...
         `)
       } else {
         gen(`
-          switch (${ch('ptr')}) {
-            case ${code('null')}:
-            parse.pointer = ptr + 4
-            return null
-            default:
-            throw new Error("Unexpected token in null")
-          }
```
Comment (mafintosh) [inline lib/null.js]: it should still return null here and return the property

File: lib/null.js [hunk 2]
```diff
@@ -12,38 +12,26 @@ function defaults (opts) {
...
           if (${ch('ptr')} === ${code('null')}) {
-            ${prop.set('null', null)}
+            ${prop.set('null', true)}
             ptr += 4
           }
         `)
       } else {
         gen(`
           if(${ch('ptr')} === ${code('null')}) {
-            parse.pointer = ptr + 4
```
Comment (mafintosh) [inline lib/null.js]: you still wanna update parse.pointer here

**Comment by mafintosh** [comment]
The fullMatch: false option is allowing the parser to make unsafe shortcuts by not fully parsing the input by expecting the input to be wellformed.

**Comment by mafintosh** [comment]
@arj03 tweaked it for you

**Comment by arj03** [comment]
Thank you :heart:

**Pull request merged**